src/CoverageController.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * Code Coverage Controller for Behat
+ *
+ * @copyright 2018 Danny Lewis
+ * @license BSD-2-Clause
+ */
+
+namespace LeanPHP\Behat\CodeCoverage;
+
+use Behat\Testwork\Cli\Controller;
+use LeanPHP\Behat\CodeCoverage\Service\ReportService;
+use SebastianBergmann\CodeCoverage\CodeCoverage;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+final class CoverageController implements Controller
+{
+
+    private $container;
+    private $coverage;
+    private $reportService;
+    private $compilerPasses;
+
+    public function __construct(ContainerBuilder $container, CodeCoverage $coverage, ReportService $reportService, $compilerPasses)
```

Comment:
The controller should not depend on container directly, but rather on specific services that it needs.